core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/TrafT4PooledConnectionManager.java

```diff
@@ -137,12 +137,18 @@ public Connection getConnection() throws SQLException {
 				}
 			} else {
 				pc = (PooledConnection) free_.get(0);
-				if (removeFreeConnection(pc, true)) {
-					TrafT4Connection c = (TrafT4Connection) pc.getConnection();
-					try {
-						c.ic_.enforceT4ConnectionTimeout(c);
+				TrafT4Connection c = (TrafT4Connection) pc.getConnection();
+				try {
+					c.ic_.enforceT4ConnectionTimeout(c);
+					if (removeFreeConnection(pc, true)) {
 						validConnection = true;
-					} catch (SQLException sqlEx) {
+					}
+				} catch (Exception e) {
+					//ignore errors
+				} finally {
+					if (!validConnection) {
+						if (free_.remove(0) != null)
```

Comment:
Thanks Selva. it was refected in https://github.com/apache/incubator-trafodion/pull/1035. Please review again.